src/affineNormL2.jl

```diff
@@ -0,0 +1,74 @@
+# Composition of the L2 norm with an affine function
+export AffineNormL2
+
+@doc raw"""
+    AffineNormL2(A,b,λ=1)
+
+Returns the ``\ell_{2}`` norm operator composed with an affine function
+```math
+f(x) = λ ||Ax+b||_2
+```
+where ``\lambda > 0``, A is some mxn matrix and b is a vector in Rᵐ.
+"""
+struct AffineNormL2{R <: Real, V1 <: AbstractMatrix{R}, V2 <: AbstractVector{R}}
+  lambda::R
+  A::V1
+  b::V2
+  function AffineNormL2{R,V1,V2}(A::V1,b::V2,lambda::R) where {R <: Real, V1 <: AbstractMatrix{R}, V2 <: AbstractVector{R}}
+    if lambda < 0
+      error("Affine Norm L2 : parameter λ must be nonnegative")
+    elseif size(A,1) != length(b)
+      error("Affine Norm L2 : dimensions of parameter A and b must match")
+    else
+      new{R,V1,V2}(lambda,A,b)
+    end
+  end
+end
+
+AffineNormL2(A::V1,b::V2,lambda::R = 1.0) where {R <: Real, V1 <: AbstractMatrix{R},V2 <: AbstractVector{R}} = AffineNormL2{R,V1,V2}(A,b,lambda)
+
+function (f::AffineNormL2)(x::AbstractVector{T}) where {T <: Real}
+  if size(A,2) != length(x)
+    error("Affine Norm L2 : dimensions of x does not match dimensions of A")
+  end
+  return f.lambda * norm(f.A*x + f.b)
+end
+
+function prox!(
+  y::AbstractVector{T},
+  f::AffineNormL2{R,V1,V2},
+  x::AbstractVector{T},
+  gamma::Real = 1,
+) where {T <: Real, R <: Real, V1 <: AbstractMatrix{R}, V2 <: AbstractVector{R}}
+
+try
+  z = -f.A*f.A'\(f.A*x + f.b)
+catch ex 
+  if isa(ex,LinearAlgebra.SingularException)
+    error("Affine Norm L2 : A is not full row rank !")
+  else
+    rethrow()
+  end
```

Comment:
`try / catch` can be quite slow. I recommend removing it. If an error occurs, we will know that it's because `A` does not have full row rank.